vernac/record.ml

```diff
@@ -883,15 +877,15 @@ let regular_structure ~cumulative ~template ~ubind ~impargs ~univs ~variances ~p
     { Data.id = name.CAst.v; idbuild; rdata; is_coercion; coers }
   in
   let data = List.map2 map data records in
-  let inds = declare_structure ~cumulative finite ~ubind ~univs ~variances
+  let inds = declare_structure ~cumulative finite ~ubind ~univs ~variances ~primitive_proj
       impargs params template data
   in
   List.map (fun ind -> GlobRef.IndRef ind) inds
 
 (** [fs] corresponds to fields and [ps] to parameters; [coers] is a
     list telling if the corresponding fields must me declared as coercions
     or subinstances. *)
-let definition_structure udecl kind ~template ~cumulative ~poly
+let definition_structure udecl kind ~template ~cumulative ~poly ~primitive_proj
```

Comment:
Unfortunately, we have to pass the `primitive_proj` bool all the way here, even though it's only relevent for classes and other structures.